package/libpiksi/libpiksi/src/settings.c

```diff
@@ -471,30 +752,163 @@ static int setting_register(settings_ctx_t *ctx, const char *section,
   }
   msg_len += l;
 
-  /* Register with daemon */
-  compare_init(ctx, msg, msg_header_len);
-
-  u8 tries = 0;
-  bool success = false;
-  do {
-    sbp_zmq_tx_send(sbp_zmq_pubsub_tx_ctx_get(ctx->pubsub_ctx),
-                    SBP_MSG_SETTINGS_REGISTER, msg_len, msg);
-    zmq_simple_loop_timeout(sbp_zmq_pubsub_zloop_get(ctx->pubsub_ctx),
-                            REGISTER_TIMEOUT_ms);
-    success = compare_match(ctx);
+  return setting_perform_request_reply(ctx,
+                                       SBP_MSG_SETTINGS_REGISTER,
+                                       msg,
+                                       msg_len,
+                                       msg_header_len,
+                                       REGISTER_TIMEOUT_MS,
+                                       REGISTER_TRIES);
+}
 
-  } while (!success && (++tries < REGISTER_TRIES));
+/**
+ * @brief setting_read_watched_value - perform SBP_MSG_SETTINGS_READ_REQ req/reply
+ * @param ctx: setting context
+ * @param setting_data: setting to read from settings daemon
+ * @return zero on success, -1 the transaction failed to complete
+ */
+static int setting_read_watched_value(settings_ctx_t *ctx, setting_data_t *setting_data)
+{
+  /* Build message */
+  u8 msg[SBP_PAYLOAD_SIZE_MAX];
+  u8 msg_len = 0;
+  int l;
 
-  compare_deinit(ctx);
+  if (!setting_data->watchonly) {
+    piksi_log(LOG_ERR, "cannot update non-watchonly setting manually");
```

Comment:
Why is this the case?

Reply:
Technically this isn't true. But the intention of the owned (non-watchonly) setting is that it should be it's own source of truth.

Reply:
Ah, OK, makes sense.